### PR TITLE
ASimWebSessionBarracudaWAF.yaml-57

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for BaracudeCEF and added BaracudaCEF parser in it. In BaracudaWAF file removed BaracudaCEF parser and in ASimWebSession added BaracudeCEF Parser name  and same changes have done for vim as well.

   Reason for Change(s):
   - To Split the BaracudaWAF and BaracudaCEF parser separetly

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please resolve below KQL validation error. Thanks
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
PR is having failure validations; please check on them. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please update on above. Thanks

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Validations failing please check. Thanks

File: Parsers/ASimWebSession/Parsers/imWebSession.yaml [hunk 1]
```diff
@@ -98,6 +99,7 @@ Parsers:
   - _Im_WebSession_FortinetFortiGate
   - _Im_WebSession_CiscoMeraki
   - _Im_WebSession_BarracudaWAF
+  - _Im_WebSession_BarracudaWAF
```
Comment (vakohl) [inline Parsers/ASimWebSession/Parsers/imWebSession.yaml:102]: _Im_WebSession_BarracudaCEF

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml:1]: Can you please check below error?
(0) Error: Missing mandatory field [Type]

File: Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
Comment (vakohl) [inline Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml:1]: Can you please check below errors?
(0) Error: Missing mandatory field [TimeGenerated]
(0) Error: Missing mandatory field [Type]

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml (hunk 2 above)
Comment (vakohl) [inline Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml:1]: Please check below and map:
(1) Warning: Empty value in 320 records (100.0%)  in mandatory field [EventEndTime] (Schema:WebSession)
(1) Warning: Empty value in 320 records (100.0%)  in mandatory field [EventStartTime] (Schema:WebSession)

**Review by vakohl — CHANGES_REQUESTED** [review]

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let barracudaSchema = datatable(
```
Comment (vakohl) [inline Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml:50]: we don't need this schema for cef table parser

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please act on reviewer's open comments. Thanks

File: Parsers/ASimWebSession/Parsers/imWebSession.yaml (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/imWebSession.yaml:102]: Done

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml (hunk 4 above)
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml:50]: Done

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@ References:
 - Title: Logs type Description in Barracuda WAF
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml:15]: Done

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Web Session ASIM filtering parser for Barracuda WAF
-  Version: '0.1'
-  LastUpdated: Jun 13 2023
+  Version: '0.2.1'
+  LastUpdated: May 21 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml:4]: Done

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+Product:
+  Name: Barracuda WAF
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml:14]: Done

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml [hunk 8]
```diff
@@ -0,0 +1,288 @@
+Parser:
+  Title: Web Session ASIM filtering parser for Barracuda CEF
+  Version: '0.2.1'
+  LastUpdated: May 21 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml:4]: Done

File: Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml [hunk 9]
```diff
@@ -15,7 +15,7 @@ References:
 - Title: Logs type Description in Barracuda WAF
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml:15]: Done

File: Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml [hunk 10]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Web Session ASIM parser for Barracuda WAF
-  Version: '0.1'
-  LastUpdated: Jun 13 2023
+  Version: '0.2.1'
+  LastUpdated:  May 21 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml:4]: Done

File: Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaCEF.yaml [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+Product:
+  Name: Barracuda WAF
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaCEF.yaml:14]: Done

File: Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaCEF.yaml [hunk 12]
```diff
@@ -0,0 +1,222 @@
+Parser:
+  Title: Web Session ASIM parser for Barracuda CEF
+  Version: '0.2.1'
+  LastUpdated: May 21 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaCEF.yaml:4]: Done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
I can see a few open comments from the reviewer. Please act on them.
image
And also, one suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button.

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please check and respond on above comments. Thanks

File: Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml (hunk 3 above)
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml:1]: not getting these errors while doing schema test

[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/ASimWebSessionBarracudaWAF.yaml:1]: AS data is not there in workspace closing this comment

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml (hunk 2 above)
Comment (Alekhya0824) [reply Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml:1]: As data is not there in workspace closing this comments

File: Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml (hunk 4 above)
Comment (vakohl) [reply Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaCEF.yaml]: @Alekhya0824  Please apply the same change to ASim file as well.
ASimWebSessionBarracudaWAF

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**